src/pybind/mgr/dashboard/frontend/src/app/shared/services/text-to-clipboard.service.ts

```diff
@@ -0,0 +1,27 @@
+import { Injectable } from '@angular/core';
+
+import { NotificationType } from '../enum/notification-type.enum';
+import { NotificationService } from './notification.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class TextToClipboardService {
+  constructor(private notificationService: NotificationService) {}
+
+  copyToClipboard(text = '') {
```

Comment:
I saw https://github.com/ceph/ceph/pull/30456/files#diff-13470465e0f7d61a295a13ab29076614 is adding a copy-to-clipboard directive. But as it's not merged yet I discussed it with @p-se. We agreed on adding the directive/service in my PR and he'll adapt his PR to it (as soon as his is ready).

Reply:
Urgh... okay. We do already have a third one :/ https://github.com/ceph/ceph/blob/master/src/pybind/mgr/dashboard/frontend/src/app/shared/directives/copy2clipboard-button.directive.ts

Reply:
I had a look at the `Copy2ClipboardButtonDirective` directive. There is one open question - in my code I'm giving the `TextToClipboardService` service a formatted string: `this.textToClipboardService.copyToClipboard(JSON.stringify(report, null, 2))`. I'm wondering if there is an easy way to archive the same with `Copy2ClipboardButtonDirective` as it takes only the control name (if I'm not wrong here). Or shouldn't be the copied string formatted at all?

Reply:
Can we please create a refactoring tracker issue to capture this proliferation of copy-to-clipboard directives? I think it would make sense to consolidate this.

Reply:
@LenzGr sure, I can create a tracker issue. I'm only wondering about what exactly as there is currently one implementation in the codebase. Do you mean I should leave my copy-to-clipboard service as it currently is and we are going to merge this and the already existing directive at a later point?

Reply:
@s0nea, what about (BTW now I see the Directive is always applied to a button, I think we should replace the directive with a component [[44960]](https://tracker.ceph.com/issues/44960)):

```html
<button ... cdCopy2ClipboardButton="report | json">
```

Reply:
Seems that the input parameter for that directive is the ID of an HTML element, so in that case just `cdCopy2ClipboardButton="report"` would work (the text inside is already formatted, right)?

Reply:
Yes, more or less. A few changes in the directive itself were needed: https://github.com/ceph/ceph/pull/33462/commits/01c4ce8745a86e895c70b90a603225a43a717631